Hazel/src/Hazel/Renderer/Renderer2D.cpp

```diff
@@ -106,6 +106,8 @@ namespace Hazel {
 	void Renderer2D::Shutdown()
 	{
 		HZ_PROFILE_FUNCTION();
+
+		delete[] s_Data.QuadVertexBufferBase;
```

Comment:
Shouldn't `Renderer2DData` delete its own memory instead? 🤔

Reply:
I agree, `Renderer2DData` should be responsible for cleaning-up it's own memory.

Reply:
Then shouldn't it create it's own resources?

Reply:
It should, In that case you can declare a constructor for `Renderer2DData` which initializes everything (technically the whole `Renderer2D::Init` function) and a destructor that frees it, convert the 
static instance (`s_Data`) to a pointer, and `new` it in the `Renderer2D::Init` function and `delete` it in `Renderer2D::Shutdown`. not sure why cherno didn't do it this way, maybe he has a good reason?  not sure, but i think it's better (__RAII__ stuff, you know).

Reply:
@IshoAntar I agree, I don't think there's a single reason why it shouldn't be that way.

Reply:
I tried, and it can't be done with a constructor, because it attempts to create OpenGL resources before OpenGL is initialized.  Perhaps Renderer2DData could have an `Init()` or `Shutdown` equivalent, maybe there is a better way?

Reply:
What about using an `Init()` and `Destroy()` pair instead of making `s_Data` a pointer, because it seems so tedious.

Reply:
I think the intention behind the `Renderer2DData` struct is just for it to store a bunch of static state data that we need. It doesn't really need its own constructor and destructor as this will just clutter the file. As @IshoAntar mentioned, it's just a container and not a full object. I think it's perfectly fine for us to manually construct and delete this one variable in the Renderer2D methods.

Reply:
I completely agree with @Puddlestomper: `Renderer2DData` is just a data struct, no functionality, and it is linked to the `Renderer2D`. The struct itself is static, and it's the `Renderer2D` that is responsible for it, so this is correct, it should be deleted on shutdown, as it is now.